src/audio/smart_amp/smart_amp.c

```diff
@@ -598,19 +598,19 @@ static int smart_amp_copy(struct comp_dev *dev)
 			avail_feedback_frames =
```

Comment:
Some style and other minor issues, can you address the checkpatch findings in https://github.com/thesofproject/sof/actions/runs/4112079634/jobs/7099324790

Reply:
@lgirdwood that's a good question.

It's difficult to tell. However during previous investigation we used to adopt smart_amp_test (SAMPLE_SMART_AMP=y) to run on ADL-P max98373 device, the pop noise was still able to be reproduced. Nevertheless, we had never met the pop noise issue on TGL max98373 devices.

Another thing which had confused me for a long time is that `smart_amp.c` sounds like to play as the role of common adapter code implementation while there exists `smart_amp_maxim_dsm.c` and `smart_amp_generic.c` for Maxim DSM and generic solution respectively. However, I then realized that  [`smart_amp.c`](https://github.com/thesofproject/sof/blob/main/src/audio/smart_amp/smart_amp.c) basically serves Maxim DSM only according that UUID is directly defined as Maxim DSM. Also, the initial author of `smart_amp.c` is from Maxim.

I wonder if we would expect smart_amp to be a generic component which may have various solution implementations (somewhat similar to module_adapter). If so, it will need additional (future) commits to partition the current codes into common and solution-specific parts.

Reply:
Good point @johnylin76 - we have named this in a generic matter, and some of the logic will probably be generic and the other logic will be specific for Maxim only. I think it would make sense in the future to split out and partition smart amp into generic and vendor files ? @cujomalainey @mwasko any thoughts ?